### PR TITLE
docs(website): add more docs

### PR DESCRIPTION
**Pull request by dionysuzx**
## summary of changes
- add concepts on creating L2 blocks (intrinsic validity checks, anchor transaction, still needs block mapping function docs but will add in subsequent PR)
- add concepts for taiko nodes
- updates all guides to have a familiar schema with troubleshooting sections
- update node running guide for simplicity
- add protocol constants to reference

@davidtaikocha after this PR is merged we can:
1. update this PR: https://github.com/taikoxyz/simple-taiko-node/pull/36 and remove the guide. instead in the README we can refer to the guide here on the website, or the "concept" page on taiko nodes.
2. delete docs from taiko-client: https://github.com/taikoxyz/taiko-client/tree/main/docs and refer to the "concepts" to the website in the README

### CODE REVIEW

**Review by dantaik — DISMISSED** [review]

**Review by dantaik — DISMISSED** [review]

File: packages/website/pages/docs/concepts/creating-l2-blocks/intrinsic-validity-functions.mdx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+| **beneficiary**         | $C_c$             | The 20-byte address to which all transaction fees in the block will be transferred.  |
+| **timestamp**           | $C_s$             | The timestamp used in the block, set to the enclosing L1 timestamp.                  |
+| **mixHash**             | $C_m$             | The mixHash value used in the block, set to the enclosing L1 mixHash.                |
+| **txListHash**          | $C_t$             | The Keccak-256 hash of this block’s txList (KZG commitment after EIP-4844).          |
+| **l1Height**            | $C_a$             | The enclosing L1 block’s parent block number.                                        |
+| **l1Hash**              | $C_h$             | The enclosing L1 block’s parent block hash.                                          |
+| **TaikoL1**             | $R$               | The TaikoL1 contract.                                                                |
+| **numProposedBlocks**   | $R_i$             | The id for the next proposed block.                                                  |
```
Comment (davidtaikocha) [inline packages/website/pages/docs/concepts/creating-l2-blocks/intrinsic-validity-functions.mdx:28]: Seems the id for the next proposed block should be `number of proposed blocks` + `1`?

File: packages/website/pages/docs/concepts/taiko-nodes.mdx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+1. Get the corresponding `TaikoL1.proposeBlock` L1 transaction.
+2. Decode the txList and block metadata from the transaction's calldata.
+3. Check whether the txList is valid based on the rules defined in Taiko protocol.
+
+If the txList is **valid**:
+
+4. Assemble a deterministic `TaikoL2.anchor` transaction based on the rules defined in the protocol, and put it as the first transaction in the proposed txList.
+5. Use this txList and the decoded block metadata to assemble a deterministic L2 block.
+6. Direct L2 execution engine to insert this assembled block and set it as the current chain's head via the Engine API.
```
Comment (davidtaikocha) [inline packages/website/pages/docs/concepts/taiko-nodes.mdx:44]:
```suggestion
6. Direct L2 execution engine to insert this assembled block and set it as the current canonical chain's head via the Engine API.
```

File: packages/website/public/images/diagrams/taiko-nodes.png
Comment (davidtaikocha) [inline packages/website/public/images/diagrams/taiko-nodes.png:1]: seems the description in taiko-client should be `beacon client`?

Comment (d1onys1us) [reply packages/website/public/images/diagrams/taiko-nodes.png:1]: [3b3871c](https://github.com/taikoxyz/taiko-mono/pull/13316/commits/3b3871c8aa54cb0b78184984857bc0f2180c7bcd)

File: packages/website/pages/docs/concepts/taiko-nodes.mdx (hunk 2 above)
Comment (d1onys1us) [reply packages/website/pages/docs/concepts/taiko-nodes.mdx:44]: [5dc2802](https://github.com/taikoxyz/taiko-mono/pull/13316/commits/5dc2802d8ee58150d79f8027abe2539e6dd532b9)

File: packages/website/pages/docs/concepts/creating-l2-blocks/intrinsic-validity-functions.mdx (hunk 1 above)
Comment (d1onys1us) [reply packages/website/pages/docs/concepts/creating-l2-blocks/intrinsic-validity-functions.mdx]: [5dc2802](https://github.com/taikoxyz/taiko-mono/pull/13316/commits/5dc2802d8ee58150d79f8027abe2539e6dd532b9)

File: packages/website/pages/docs/concepts/creating-l2-blocks/intrinsic-validity-functions.mdx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+| **timestamp**           | $C_s$             | The timestamp used in the block, set to the enclosing L1 timestamp.                                     |
+| **mixHash**             | $C_m$             | The mixHash value used in the block, set to the enclosing L1 mixHash.                                   |
+| **txListHash**          | $C_t$             | The Keccak-256 hash of this block’s txList (KZG commitment after EIP-4844).                             |
+| **l1Height**            | $C_a$             | The enclosing L1 block’s parent block number.                                                           |
+| **l1Hash**              | $C_h$             | The enclosing L1 block’s parent block hash.                                                             |
+| **TaikoL1**             | $R$               | The TaikoL1 contract.                                                                                   |
+| **numProposedBlocks**   | $R_i$             | The current number of proposed blocks (he `id` for the next proposed block is `numProposedBlocks + 1`). |
```
Comment (davidtaikocha) [inline packages/website/pages/docs/concepts/creating-l2-blocks/intrinsic-validity-functions.mdx:28]:
```suggestion
| **numProposedBlocks**   | $R_i$             | The current number of proposed blocks (The `id` for the next proposed block is `numProposedBlocks + 1`). |
```

**Review by davidtaikocha — APPROVED** [review]

**Review by dantaik — APPROVED** [review]

**Pull request merged**